### PR TITLE
Extract toSttp converter which returns decode results

### PR DESCRIPTION
**Pull request by ghostbuster91**

### CODE REVIEW

File: client/sttp-client/src/main/scala/tapir/client/sttp/EndpointToSttpClient.scala [hunk 1]
```diff
@@ -11,48 +11,60 @@ import tapir.internal._
 
 class EndpointToSttpClient(clientOptions: SttpClientOptions) {
   def toSttpRequest[I, E, O, S](e: Endpoint[I, E, O, S], baseUri: Uri): I => Request[Either[E, O], S] = { params =>
-    val (uri, req1) =
-      setInputParams(e.input.asVectorOfSingleInputs, paramsTupleToParams(params), 0, baseUri, basicRequest.asInstanceOf[PartialAnyRequest])
+    toSttpRequestRaw(e, baseUri)(params).mapResponse(_.bimap(getOrThrow, getOrThrow))
+  }
 
-    val req2 = req1.copy[Identity, Any, Any](method = sttp.model.Method(e.input.method.getOrElse(Method.GET).method), uri = uri)
+  def toSttpRequestRaw[S, O, E, I](e: Endpoint[I, E, O, S], baseUri: Uri): I => Request[Either[DecodeResult[E], DecodeResult[O]], S] = {
```
Comment (adamw) [inline client/sttp-client/src/main/scala/tapir/client/sttp/EndpointToSttpClient.scala]: why not `Request[DecodeResult[Either[E, O]]`? :)

**Comment by ghostbuster91** [comment]
Related to #244

File: client/sttp-client/src/main/scala/tapir/client/sttp/EndpointToSttpClient.scala [hunk 2]
```diff
@@ -11,8 +11,18 @@ import tapir.internal._
 
 class EndpointToSttpClient(clientOptions: SttpClientOptions) {
   def toSttpRequest[I, E, O, S](e: Endpoint[I, E, O, S], baseUri: Uri): I => Request[Either[E, O], S] = { params =>
+    toSttpRequestRaw(e, baseUri)(params).mapResponse(getOrThrow)
+  }
+
+  def toSttpRequestRaw[S, O, E, I](e: Endpoint[I, E, O, S], baseUri: Uri): I => Request[DecodeResult[Either[E, O]], S] = { params =>
```
Comment (adamw) [inline client/sttp-client/src/main/scala/tapir/client/sttp/EndpointToSttpClient.scala]: I think we need a method in the package object as well? + some comment explaining the difference (or rather, specifying that the `toSttpRequest` method can throw an exception).

I'm also thinking about naming ... Not sure about the `raw` :) Maybe `safe`?

Comment (ghostbuster91) [reply client/sttp-client/src/main/scala/tapir/client/sttp/EndpointToSttpClient.scala]: How about now?

**Comment by adamw** [comment]
Thanks :)

**Pull request merged**